### PR TITLE
[WIP] AWQ initial implementation

### PR DESCRIPTION
**Pull request by rahul-tuli**
SUMMARY:
"please provide a brief summary"

TEST PLAN:
"please outline how the changes were tested"

### CODE REVIEW

**Comment by qingquansong** [comment]
Hey @rahul-tuli , any timeline about when AWQ will be shipped and merged? Thank you!

File: src/llmcompressor/modifiers/quantization/awq/base.py [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+        pass  # nothing needed for this modifier
+
+    def on_initialize(self, state: State, **kwargs) -> bool:
+        """
+        Initialize and run AWQ on the given state
+
+        :param state: state to run AWQ on
+        :return: True on a successful run, False otherwise
+        """
+        if self.end and self.end != -1:
```
Comment (kylesayrs) [inline src/llmcompressor/modifiers/quantization/awq/base.py:111]: Currently user can pass self.end == 0
```suggestion
        if self.end is not None or self.end == -1:
```

File: src/llmcompressor/modifiers/quantization/awq/base.py [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+
+        :param model: model to apply smoothing to
+        """
+        logger.info("Smoothing activation scales...")
+        for mapping in self.resolved_mappings_:
+            smooth_layer = mapping.smooth_layer
+            balance_layers = mapping.balance_layers
+
+            activations = self.scales_[mapping.smooth_name].inps
+            module2inspect = smooth_layer
```
Comment (kylesayrs) [inline src/llmcompressor/modifiers/quantization/awq/base.py:273]: Not sure why there are two variable names

File: src/llmcompressor/modifiers/quantization/awq/base.py [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+            # each of the quantization groups, and rescales each group
+            # individually so that each group has weights on a 0-1 scale.
+            w_scale = weight.abs() / (weight.abs().amax(dim=1, keepdim=True) + 1e-6)
+            # Resizes the rescaled weight matrix back up to its original dimensions
+            w_scale = w_scale.view(org_shape)
+            # Gets the average rescaled magnitude for each output channel
+            w_mean = w_scale.mean(0)
+
+            # [STEP 2]: Compute per-channel mean of the input activation with chunking
+            # move inp to cpu to avoid memory leak
```
Comment (kylesayrs) [inline src/llmcompressor/modifiers/quantization/awq/base.py:291]: What operation causes the memory leak? Should the tensor be moved back to gpu after?

**Comment by rahul-tuli** [comment]
> Hey @rahul-tuli , any timeline about when AWQ will be shipped and merged? Thank you!

Soon!

**Pull request closed without merging**